### PR TITLE
Document the ability to specify repair steps in appinfo/info.xml

### LINKED ISSUE #390: [Developer Docs] repair steps for app migration is now in info.xml
Three new entries in `info.xml` can be used to execute repair steps:

- Before database migration
- After database migration
- As a background job to be executed asynchronously after migration, in live operations
- Install and uninstall steps

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #390.

It:

- updates the comprehensive appinfo/info.xml example to include the repair-steps configuration
- documents the repair-steps options

It's slightly larger than it needs to be, as the comprehensive appinfo/info.xml example was extracted out into a partial (so that it can be linted, validated, etc). Otherwise, only changes strictly related to #390 have been made in this change.

### CODE REVIEW

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        <lib min-version="1.2">xml</lib>
+        <lib max-version="2.0">intl</lib>
+        <lib>curl</lib>
+        <os>Linux</os>
+        <owncloud min-version="6.0.4" max-version="8"/>
+    </dependencies>
+
+	<repair-steps>
+		<install>
+			<step>OCA\FilesEncryption\Migration\Install</step>
```
Comment (DeepDiver1975) [inline modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: this sample should be consistent - name space is 'YourapplicationsNamespace' - so this should be OCA\YourapplicationsNamespace\...

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			<step>OCA\FilesEncryption\Migration\LiveMigration</step>
+		</live-migration>
+		<uninstall>
+			<step>OCA\FilesEncryption\Migration\Uninstall</step>
+		</uninstall>
+	</repair-steps>
+
+    <!-- For registering panels -->
+    <settings>
+        <admin>OCA\MyApp\Settings\Admin</admin>
```
Comment (DeepDiver1975) [inline modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: namespace shall be in sync with the rest of the app

**Comment by settermjd** [comment]
@DeepDiver1975, any additional changes required?

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <!-- Category values available at: https://marketplace.owncloud.com/ajax/categories -->
+    <category>A category for the application. </category>
+    <summary>A summary of the application's purpose (max 90 chars)</summary>
+
+    <types>
+        <filesystem/>
+    </types>
+
+    <documentation>
+        <user>https://doc.owncloud.com/server/latest/user_manual/pim/contacts.html</user>
```
Comment (DeepDiver1975) [inline modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: do there urls exist?

Comment (settermjd) [reply modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: Does the URL exist? Yes.

**Comment by settermjd** [comment]
@DeepDiver1975, is this now ready to merge?

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml (hunk 3 above)
Comment (DeepDiver1975) [reply modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: No idea if we should reference the docs of he contacts app for this example app being named yourapp .... no idea - up to you @settermjd

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        <admin>https://doc.owncloud.com/server/latest/admin_manual/configuration_server/occ_command.html?highlight=contact#dav-commands</admin>
+        <developer>https://github.com/owncloud/contacts/blob/master/README.md</developer>
+    </documentation>
+
+    <author>Your Name</author>
+    <namespace>YourapplicationsNamespace</namespace>
+    <website>https://owncloud.org</website>
+    <bugs>https://github.com/owncloud/theapp/issues</bugs>
+    <repository type="git">https://github.com/owncloud/theapplication.git</repository>
+    <ocsid>1234</ocsid>
```
Comment (DeepDiver1975) [inline modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: kill this - long gone

File: modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    <documentation>
+        <user>https://doc.owncloud.com/server/latest/user_manual/pim/contacts.html</user>
+        <admin>https://doc.owncloud.com/server/latest/admin_manual/configuration_server/occ_command.html?highlight=contact#dav-commands</admin>
+        <developer>https://github.com/owncloud/contacts/blob/master/README.md</developer>
+    </documentation>
+
+    <author>Your Name</author>
+    <namespace>YourapplicationsNamespace</namespace>
+    <website>https://owncloud.org</website>
+    <bugs>https://github.com/owncloud/theapp/issues</bugs>
```
Comment (DeepDiver1975) [inline modules/developer_manual/pages/_partials/app/fundamentals/complete-info.xml]: yourapp - thisapp - theapplication - this should be consistent

**Pull request merged**